packages/react-components/react-card/src/components/CardHeader/useCardHeaderStyles.ts

```diff
@@ -8,46 +8,33 @@ import type { CardHeaderSlots, CardHeaderState } from './CardHeader.types';
 export const cardHeaderClassNames: SlotClassNames<CardHeaderSlots> = {
   root: 'fui-CardHeader',
   image: 'fui-CardHeader__image',
-  content: 'fui-CardHeader__content',
   header: 'fui-CardHeader__header',
   description: 'fui-CardHeader__description',
   action: 'fui-CardHeader__action',
 };
 
 const useStyles = makeStyles({
   root: {
-    display: 'flex',
-    flexDirection: 'row',
+    display: 'grid',
+    gridAutoColumns: 'min-content 1fr min-content',
     alignItems: 'center',
-    ...shorthands.gap('12px'),
-    height: '32px',
+    ...shorthands.gap('12px', '0'),
   },
   image: {
-    minWidth: '24px',
-    minHeight: '24px',
-    maxWidth: '32px',
-    maxHeight: '32px',
-
-    display: 'flex',
-    alignItems: 'center',
-
-    '> *': {
-      minWidth: 'inherit',
-      minHeight: 'inherit',
-      maxWidth: '100%',
-      maxHeight: '100%',
-    },
+    gridColumnStart: '1',
+    gridRowStart: 'span 2',
   },
-
-  textContainer: {
-    display: 'flex',
-    flexDirection: 'column',
-    flexGrow: 1,
-    height: 'inherit',
-
-    '> *': {
-      height: '50%',
-    },
+  header: {
+    gridColumnStart: '2',
+    gridRowStart: '1',
+  },
+  description: {
```

Comment:
q: so what's gonna happen if user omits header, based on this there will be blank row correct? Is that intended ?

Reply:
That's a good point, let me double check the design templates

Reply:
From the design side, CardHeader only requires a `title` and should support all other possible scenarios. Leveraging `grid-auto-rows` with `min-content` we can get the behavior we're looking for as the `description` content will be collapsed.
However, `grid-auto-columns` and not providing an `image` does not seem to work in a similar way. I'll need to research a bit more to guarantee all scenarios pass and will add a vr test covering this.

Reply:
I've opted out of using `gap` to counteract the weird `grid` interaction on empty columns. `margin` is now used to space the elements.